src/target/riscv/riscv.c

```diff
@@ -506,27 +506,48 @@ static int find_trigger(struct target *target, int type, bool chained, int *idx)
 	return ERROR_FAIL;
 }
 
+static int find_first_trigger_by_id(struct target *target, int unique_id)
+{
+	RISCV_INFO(r);
+
+	for (unsigned i = 0; i < r->trigger_count; i++) {
+		if (r->trigger_unique_id[i] == unique_id)
+			return i;
+	}
+	return -1;
+}
+
 static int set_trigger(struct target *target, int idx, riscv_reg_t tdata1, riscv_reg_t tdata2,
 	riscv_reg_t tdata1_ignore_mask)
 {
-	riscv_reg_t tdata1_rb;
+	riscv_reg_t tdata1_rb, tdata2_rb;
 	if (riscv_set_register(target, GDB_REGNO_TSELECT, idx) != ERROR_OK)
 		return ERROR_FAIL;
 	if (riscv_set_register(target, GDB_REGNO_TDATA1, tdata1) != ERROR_OK)
 		return ERROR_FAIL;
 	if (riscv_get_register(target, &tdata1_rb, GDB_REGNO_TDATA1) != ERROR_OK)
 		return ERROR_FAIL;
 	if ((tdata1 & ~tdata1_ignore_mask) != (tdata1_rb & ~tdata1_ignore_mask)) {
-		LOG_DEBUG("Trigger doesn't support what we need; After writing 0x%"
+		LOG_TARGET_DEBUG(target,
+			"Trigger doesn't support what we need; After writing 0x%"
 			PRIx64 " to tdata1 it contains 0x%" PRIx64
 			"; tdata1_ignore_mask=0x%" PRIx64,
 			tdata1, tdata1_rb, tdata1_ignore_mask);
```

Comment:
Minor thing: For easier troubleshooting, it will be good to have the trigger index in the debug message.

Also the `idx` parameter in `set_trigger()` (and in `find_trigger()`) can be made `unsigned`.

```suggestion
		LOG_TARGET_DEBUG(target,
			"Trigger #%u doesn't support what we need; After writing 0x%"
			PRIx64 " to tdata1 it contains 0x%" PRIx64
			"; tdata1_ignore_mask=0x%" PRIx64,
			idx, tdata1, tdata1_rb, tdata1_ignore_mask);
```

Reply:
Done.